_posts/2022-04-11-changelog-168.markdown

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: "IntelliJ Rust Changelog #168"
+date: 2022-04-11 13:00:00 +0300
+---
+
+
+## New Features
+
+* [#8647] Collapse macro expansion steps into single one.
+It's not only fixes UI blinking during macro expansion process but also significantly speed up macro expansion itself
+
+* [#8689] Provide file links in terminal for `dbg!` macro output
+
+  {% include gif-img.html path="/assets/posts/changelog-168/dbg_output_file_links" w="700px" %}
+
+## Fixes
+
+* [#8674] The `Generate constructor` action now places the cursor after the generated constructor (by [@Kobzol])
+
+* [#8671] Escape keyword-like identifiers like `type` by `Import` quick-fix
+
+* [#8631] Properly evaluate build scripts even if a crate doesn't contain binary or library targets.
+  Note, build script evaluation is disabled by default. To turn it on, enable `org.rust.cargo.evaluate.build.scripts` [experimental feature][experimental features]
+
+* [#8692] Update [ra_ap_proc_macro_srv](https://crates.io/crates/ra_ap_proc_macro_srv) lib to 0.0.102 in procedural macro expander.
+  It should fix expansion of some procedural macro calls (for example, from [rocket](https://crates.io/crates/rocket) crate).
+  Note, procedural macro expansion is disabled by default for now.
+  To turn it on, enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` [experimental features]
+
+* [#8666] Fix `NoClassDefFoundError` exceptions in Rider 2021.3 and show debugger settings properly
+
+* [#8683] Properly validate new Rust project names with `.` symbol in minor IDEs like CLion
```

Comment:
```suggestion
* [#8683] Properly validate project names containing `.` symbol while creating new Rust projects in all IntelliJ-based IDEs
```

Reply:
I'm not sure about "all IntelliJ-based IDEs". But "minor IDEs" term looks confusing to me

Reply:
It's definetly not about all IntelliJ-based IDEs because it doesn't affect IDEA
Note, we already used `minor IDE` term several times in previous changelogs

Reply:
Yes, I understand that "IntelliJ-based IDEs" term is not precise in this context.

I've seen "minor IDE" term several times in both our GitHub and JetBrains YouTrack. But this time, I took a fresh look at it and found it confusing. Since I don't see a good alternative too, probably this question is better to address to a certain team like copywriters